tools/profileBuilder/model/transforms.go

```diff
@@ -35,155 +35,127 @@ import (
+	Include []string `json:"include"`
 }
 
-const armPathModifier = "mgmt"
+const (
+	armPathModifier = "mgmt"
+	aliasFileName   = "models.go"
+)
 
-var packageName = regexp.MustCompile(`services[/\\](?P<provider>[\w\-\.\d_\\/]+)[/\\](?:(?P<arm>` + armPathModifier + `)[/\\])?(?P<version>v?\d{4}-\d{2}-\d{2}[\w\d\.\-]*|v?\d+\.\d+[\.\d\w\-]*)[/\\](?P<group>[/\\\w\d\-\._]+)`)
+var packageName = regexp.MustCompile(`services[/\\](?P<provider>[\w\-\.\d_\\/]+)[/\\](?:(?P<arm>` + armPathModifier + `)[/\\])?(?P<version>v?\d{4}-\d{2}-\d{2}[\w\d\.\-]*|v?\d+[\.\d+\.\d\w\-]*)[/\\](?P<group>[/\\\w\d\-\._]+)`)
 
 // BuildProfile takes a list of packages and creates a profile
-func BuildProfile(packageList collection.Enumerable, name, outputLocation string, outputLog, errLog *log.Logger) {
-	var packages collection.Enumerator
-
-	// Find the names of all of the packages for inclusion in this profile.
-	packages = packageList.Enumerate(nil).Select(func(x interface{}) interface{} {
-		if cast, ok := x.(string); ok {
-			return cast
-		}
-		return nil
-	})
-
-	// Parse the packages that were selected for inclusion in this profile.
-	packages = packages.SelectMany(func(x interface{}) collection.Enumerator {
-		results := make(chan interface{})
-
-		go func() {
-			defer close(results)
-
-			cast, ok := x.(string)
-			if !ok {
-				return
+func BuildProfile(packageList ListDefinition, name, outputLocation string, outputLog, errLog *log.Logger) {
+	wg := &sync.WaitGroup{}
+	wg.Add(len(packageList.Include))
+	for _, pkgDir := range packageList.Include {
+		if !filepath.IsAbs(pkgDir) {
+			abs, err := filepath.Abs(pkgDir)
+			if err != nil {
+				errLog.Fatalf("failed to convert to absolute path: %v", err)
 			}
-			files := token.NewFileSet()
-			parsed, err := parser.ParseDir(files, cast, nil, 0)
+			pkgDir = abs
+		}
+		go func(pd string) {
+			fs := token.NewFileSet()
+			packages, err := parser.ParseDir(fs, pd, func(f os.FileInfo) bool {
+				// exclude test files
+				return !strings.HasSuffix(f.Name(), "_test.go")
+			}, 0)
 			if err != nil {
-				errLog.Printf("Couldn't open %q because: %v", cast, err)
-				return
+				errLog.Fatalf("failed to parse '%s': %v", pd, err)
 			}
-
-			for _, entry := range parsed {
-				results <- entry
+			if len(packages) < 1 {
+				errLog.Fatalf("didn't find any packages in '%s'", pd)
 			}
-		}()
-
-		return results
-	})
-
-	packages = GenerateAliasPackages(packages, name, outputLog, errLog)
-	products := packages.ParallelSelect(GetAliasWriter(outputLocation, outputLog, errLog))
-
-	generated := 0
-
-	// Write each aliased package that was found
-	for entry := range products {
-		if entry.(bool) {
-			generated++
-		}
+			if len(packages) > 1 {
+				errLog.Fatalf("found more than one package in '%s'", pd)
+			}
+			for pn := range packages {
+				p := packages[pn]
+				// trim any non-exported nodes
+				if exp := ast.PackageExports(p); !exp {
+					errLog.Fatalf("package '%s' doesn't contain any exports", pn)
+				}
```

Comment:
i guess we fail here because it should never happen ?

Reply:
Correct.